File: 0118-pascals-triangle/main.js
var generate = function(numRows) {
  let result = [];
  for (let i = 0; i < numRows; i++) {
    let newRow = [];
    let prevRow = result[i - 1]; // undefined
    for (let j = 0; j <= i; j++) {
      if (j === 0 || j === i) {
        newRow.push(1);
      } else if (prevRow) {
        newRow.push(prevRow[j - 1] + prevRow[j]);
      }
    }
    result.push(newRow);
  }
  return result;
}

// console.log(generate(6))

File: 0118-pascals-triangle/main.go
package main

import ("fmt")

func generate(numRows int) [][]int {
  var result [][]int
	for i:= 0; i < numRows; i++ {
    var newRow []int
		for j := 0; j <= i; j++ {
			if j == 0 || j == i {
				newRow = append(newRow, 1) // push
				} else {
				prevRow := result[i - 1]
				newVal := prevRow[j - 1] + prevRow[j]
				newRow = append(newRow, newVal) // push
			}
		}
		result = append(result, newRow) // push
	}
	return result
}

func main () {
  fmt.Println(generate(6))
}